src/parser/grammar.lalrpop

```diff
@@ -863,15 +863,15 @@ NOpPre<ArgRule>: UniTerm = {
 }
 
 TypeBuiltin: Types = {
-    "Dyn" => Types(TypeF::Dyn),
-    "Num" => Types(TypeF::Num),
-    "Bool" => Types(TypeF::Bool),
-    "Str" => Types(TypeF::Str),
+     "Dyn" => Types::from(TypeF::Dyn),
+     "Num" => Types::from(TypeF::Num),
+     "Bool" => Types::from(TypeF::Bool),
+     "Str" => Types::from(TypeF::Str),
 }
 
 TypeAtom: Types = {
     <TypeBuiltin>,
-    "[|" <rows:(<EnumTag> ",")*> <last: (<EnumTag>)?> <tail: (";" <Ident>)?> "|]" => {
+    <l: @L> "[|" <rows:(<EnumTag> ",")*> <last: (<EnumTag>)?> <tail: (";" <Ident>)?> "|]" <r: @R> => {
```

Comment:
Once again, is there any reason to handle position manually inside `TypeAtom`, instead of wrapping usages with `WithPos` ? :sweat_smile: